### PR TITLE
Single course outline integration

### PR DESCRIPTION
**Pull request by nunibaranes**
- improve the data flow in AnalyticsChart child components.
- Adds properties to useCourseContext for courseOutline.
- changed call to SDK for getting correct data.
- Adds the logic for searching in courseOutline in useCourseContext.
- Updated version to 0.9.0

TODO:
- verify the data results with Eli.
- fix skeleton behavior

### CODE REVIEW

File: src/components/common/charts/WMHorizontalBarChart/style.module.scss [hunk 1]
```diff
@@ -27,6 +27,21 @@
...
+    font-family: var(--font-primary);
+    font-size: 12px;
+    font-weight: 500;
+    letter-spacing: 0;
+    display: block;
+    white-space: nowrap;
+    overflow: hidden;
+    text-overflow: ellipsis;
+    -o-text-overflow: ellipsis;
+    -moz-binding: url(<q>assets/xml/ellipsis.xml#ellipsis</q>);
```
Comment (dr15) [inline src/components/common/charts/WMHorizontalBarChart/style.module.scss:42]: Pretty sure we have autoprefixer so I would remove these

File: src/components/common/charts/WMHorizontalBarChart/index.tsx [hunk 2]
```diff
@@ -41,12 +41,35 @@ const YAxisTick = ({
...
+        className={classes['answer-option']}
+        x={x}
+        y={y}
+        dominantBaseline="middle"
+        textAnchor="end"
+      >
         {isCorrect && <tspan className={classes['check-mark']}>✔ </tspan>}
-        {value}
       </text>
+      <foreignObject
```
Comment (dr15) [inline src/components/common/charts/WMHorizontalBarChart/index.tsx:62]: Awesome!

File: src/components/common/charts/WMHorizontalBarChart/index.tsx [hunk 3]
```diff
@@ -41,12 +41,35 @@ const YAxisTick = ({
 }) => {
   const isCorrect = bars[index].isCorrect;
 
+  /**
+   * using foreignObject to add text style manipulation
+   * https://developer.mozilla.org/en-US/docs/Web/SVG/Element/foreignObject
+   *
+   * TODO: add support to Internet Explorer
```
Comment (dr15) [inline src/components/common/charts/WMHorizontalBarChart/index.tsx:48]: No need! You can remove this comment :)

File: src/components/common/AnalyticsCharts/CourseTimeCompletionChart/index.tsx [hunk 4]
```diff
@@ -17,18 +17,20 @@ export default function CoursesTimeCompletionChart({
...
   useEffect(() => {
     if (overview?.completion_time) {
       const { completion_time } = overview;
 
-      setCompletionTimeAvg(completion_time.avg ? completion_time.avg.toFixed(0) : 0);
+      setCompletionTimeAvg(completion_time.avg ? parseInt(completion_time.avg.toFixed(0)) : 0);
 
-      if (completion_time.buckets.length)
+      if (completion_time.buckets.length) {
+        console.log('completion_time.buckets ', completion_time.buckets);
```
Comment (dr15) [inline src/components/common/AnalyticsCharts/CourseTimeCompletionChart/index.tsx:31]: console.log

File: src/components/Screen/CourseScreen/CourseTabs.tsx [hunk 5]
```diff
@@ -49,7 +68,7 @@ export default function CourseTabs({ course }: ICourseTabs): ReactElement {
                 <>
                   {icon}
                   <span>
-                    {title} {itemsLength && `(${itemsLength})`}
+                    {title} {itemsLength ? `(${itemsLength})` : '(0)'}
```
Comment (dr15) [inline src/components/Screen/CourseScreen/CourseTabs.tsx:71]: Wouldn't `itemsLength` be 0 anyway?

File: src/components/Screen/CourseScreen/CourseScreenHeader.tsx [hunk 6]
```diff
@@ -42,7 +42,7 @@ export default function CourseScreenHeader({
                 variant={ButtonVariantEnum.Secondary}
                 className={classes['edit']}
               >
-                <Link to="/new-course">Edit</Link>
+                <Link to={`/course-editor/${course.id}`}>Edit</Link>
```
Comment (dr15) [inline src/components/Screen/CourseScreen/CourseScreenHeader.tsx:45]: I think there is a const for this path in the consts file, if not I would add it there

File: src/components/Screen/CourseScreen/CourseOutlineTable.tsx [hunk 7]
```diff
@@ -1,57 +1,50 @@
...
         <ExportButton className={classes['export']} />
         <SearchFilter
           className={classes['search']}
           placeholder="Search item name"
           onSearch={onSearch}
+          value={courseOutlineSearchValue}
         />
       </ControlsWrapper>
-    </WMTable>
+      {filteredCourseOutline.length ? (
```
Comment (dr15) [inline src/components/Screen/CourseScreen/CourseOutlineTable.tsx:37]: Please leave a comment about this solution, thanks :)

File: src/components/Screen/CourseScreen/CourseTabs.tsx (hunk 5 above)
Comment (nunibaranes) [reply src/components/Screen/CourseScreen/CourseTabs.tsx:71]: Yes, you right, I fixed it, the previous ternary returns 0 without parenthesis. I fixed it

**Pull request merged**